### PR TITLE
enh(resource-status): Reset page when searching

### PR DESCRIPTION
**Pull request by bdauria**
## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 20.10.x (master)

### CODE REVIEW

**Review by jdelpierre — APPROVED** [review]

File: www/front_src/src/Resources/Listing/useLoadResources.ts [hunk 1]
```diff
@@ -115,6 +116,10 @@ const useLoadResources = (): LoadResources => {
     serviceGroups,
   ]);
 
+  React.useEffect(() => {
+    setPage(1);
+  }, [currentSearch]);
```
Comment (kduret) [inline www/front_src/src/Resources/Listing/useLoadResources.ts:121]: all filters change should redirect to page 1, aren't they ?
sorting props also

File: www/front_src/src/Resources/Listing/useLoadResources/index.ts [hunk 2]
```diff
@@ -97,15 +98,22 @@ const useLoadResources = (): LoadResources => {
   }, []);
 
   React.useEffect(() => {
-    if (not(equals(currentSearch, nextSearch))) {
+    if (isNil(page)) {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/useLoadResources/index.ts:101]: Why `page` could be `null` ?

**Review by Thebarda — APPROVED** [review]

File: www/front_src/src/Resources/Listing/index.tsx [hunk 3]
```diff
@@ -107,7 +107,7 @@ const ResourceListing = (): JSX.Element => {
       loading={loading}
       columnConfiguration={columns}
       tableData={listing?.result}
-      currentPage={page - 1}
+      currentPage={(page || 1) - 1}
```
Comment (jdelpierre) [inline www/front_src/src/Resources/Listing/index.tsx:110]: why `-1` ? 
page => 1 - 1 = 0, so it's possible to have 0 value ?

**Review by kduret — APPROVED** [review]

File: www/front_src/src/Resources/Listing/index.tsx (hunk 3 above)
Comment (bdauria) [reply www/front_src/src/Resources/Listing/index.tsx:110]: pages are 0 based in the Listing

Comment (jdelpierre) [reply www/front_src/src/Resources/Listing/index.tsx:110]: ok :+1:

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**